sentry/src/main/java/io/sentry/jvmti/LocalsCache.java

```diff
@@ -0,0 +1,39 @@
+package io.sentry.jvmti;
+
+/**
+ * Utility class used by the Sentry Java Agent (https://github.com/getsentry/sentry-java-agent) to
+ * store local variable information for the last thrown exception.
+ */
+public final class LocalsCache {
+    private static ThreadLocal<Frame[]> result = new ThreadLocal<Frame[]>() {
+        @Override
+        protected Frame[] initialValue() {
+            return new Frame[0];
+        }
+    };
+
+    /**
+     * Utility class, no public ctor.
+     */
+    private LocalsCache() {
+
+    }
+
+    /**
+     * Store the local variable information for the last exception thrown on this thread.
+     *
+     * @param frames Array of {@link Frame}s to store
+     */
+    public static void setCache(Frame[] frames) {
```

Comment:
`setFrames` or something seems better in my opinion. Since this is actually caching `Frame` objects, maybe it should actually be called `FrameCache`? I think that would make `SentryStackTraceElement.fromStackTraceElements` read clearer since any case of `localsCache[i]` would be clearer that it is referencing a frame and not a collection of local variables.